### PR TITLE
Fix invalid streams for profiling hooks

### LINKED ISSUE #40: after_epoch_profile gets invalid streams
`after_epoch_profile` hook call is given invalid streams for `emloop eval`. E.g., for `emloop eval valid`, it actually gets `streams=['train']`.

This is the line to blame: https://github.com/iterait/emloop/blob/dev/emloop/main_loop.py#L261
It should use the function argument `streams` instead of `[self._train_stream_name] + self._extra_streams`.

Regression test is necessary when submitting a pull request.

@bedapisl, have you already looked into it? Maybe @gdynusa would be interested if not.

### PR DESCRIPTION
**Pull request by bedapisl**
Fixes #40.

### CODE REVIEW

**Comment by FloopCZ** [comment]
Thank you @bedapisl, this looks good. We need to add regression test though. @gdynusa Are you interested?

File: emloop/tests/cli/common_test.py [hunk 1]
```diff
@@ -343,3 +355,20 @@ def test_config_file_is_incorrect(tmpdir, caplog):
...
     run(config=config, output_root=tmpdir)
     assert 'Extra arguments: {\'non-existent\': \'none\'}' not in caplog.text
+
+
+def test_run_with_eval_stream(tmpdir, caplog):
+    """Test that eval is called with given stream and not with EL_DEFAULT_TRAIN_STREAM."""
+    caplog.set_level(logging.INFO)
+
+    config = {'dataset': {'class': 'emloop.tests.cli.common_test.DummyEvalDataset'},
+              'stream': {'train': {'rotate': 20}},
```
Comment (FloopCZ) [inline emloop/tests/cli/common_test.py]: `stream` is kind of unusual config item, what is it supposed to mean? Did you mean `eval` ?

File: emloop/tests/cli/common_test.py [hunk 2]
```diff
@@ -299,7 +311,7 @@ def test_config_file_is_incorrect(tmpdir, caplog):
 
     # incorrect dataset arguments
     config = {'dataset': {'class': 'emloop.tests.cli.common_test.DummyDataset', 'output_dir': '/tmp'},
-              'hooks': [{'emloop.tests.cli.common_test.DummyHook': {'additional_arg': 10}, 'StopAfter': {'epochs': 1}}],
+              'hooks': [{'emloop.tests.cli.common_test.DummyHook': {'additional_arg': 1}}, {'StopAfter': {'epochs': 1}}],
```
Comment (FloopCZ) [inline emloop/tests/cli/common_test.py:322]: Nice catch!

File: emloop/tests/cli/common_test.py [hunk 3]
```diff
@@ -343,3 +355,20 @@ def test_config_file_is_incorrect(tmpdir, caplog):
...
+              'stream': {'train': {'rotate': 20}},
+              'hooks': [{'LogProfile': {'epochs': 1}}, {'StopAfter': {'epochs': 1}}],
+              'model': {'class': 'emloop.tests.cli.common_test.DummyModel', 'io': {'in': ['a'], 'out': ['dummy']}}}
+
+    run(config=config, output_root=tmpdir, eval='valid')
+    
+    assert f'Running the evaluation of stream `{EL_DEFAULT_TRAIN_STREAM}`' not in caplog.text
+    assert f'T {EL_DEFAULT_TRAIN_STREAM}:' not in caplog.text
+    assert 'Running the evaluation of stream `valid`' in caplog.text
+    assert 'T valid:' in caplog.text
```
Comment (FloopCZ) [inline emloop/tests/cli/common_test.py]: I would prefer a different way. It is kind of dirty to check the log text for its contents. Can you add a dummy hook with `after_epoch_profile` method, that simply asserts that the provided `streams` parameter is `['eval']` ?

**Comment by blazekadam** [comment]
Looks good to me now. @FloopCZ merge it if you want.

**Review by blazekadam — APPROVED** [review]

File: emloop/tests/cli/common_test.py [hunk 4]
```diff
@@ -57,6 +70,16 @@ def __init__(self, variables: List[str], **kwargs):
         variables[0], variables[1] = variables[1], variables[0]
 
 
+class DummyEvalHook(AbstractHook):
+    """Dummy hook which save its ``**kwargs`` to ``self.kwargs``."""
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
```
Comment (FloopCZ) [inline emloop/tests/cli/common_test.py]: This is the default behavior. Please remove the constructor.

**Review by FloopCZ — COMMENTED** [review]
I am almost satisfied. :-)
Thank you @gdynusa for the improvement, please do one more and we can merge it.

File: emloop/tests/cli/common_test.py [hunk 5]
```diff
@@ -57,6 +70,13 @@ def __init__(self, variables: List[str], **kwargs):
         variables[0], variables[1] = variables[1], variables[0]
 
 
+class DummyEvalHook(AbstractHook):
+    """Dummy hook which save its ``**kwargs`` to ``self.kwargs``."""
```
Comment (FloopCZ) [inline emloop/tests/cli/common_test.py]:
```suggestion
    """Dummy hook that checks the `after_epoch_profile` is evaluated on `valid` stream."""
```

**Review by FloopCZ — APPROVED** [review]
Thank you @gdynusa, please merge after accepting the suggestion.

**Pull request merged**